File: RedSheet.DbEntities/Models/BackgroundEvent.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("BackgroundEvents", Schema = "dbo")]
	[Serializable]
	public partial class BackgroundEvent  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int BackgroundEventId { get; set; } 
 
	[MaxLength(1000)]
	public string Description { get; set; } 

	public Nullable<System.DateTime> EndDate { get; set; } 
 
	[Required]
	public bool IsEvent { get; set; } 
 
	[Required]
	public System.DateTime StartDate { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Title { get; set; } 
 
	[RelationshipTableAttribue("ProjectBackgrounds","dbo")]
	//Relationships
	public int ProjectBackgroundId { get; set; } 
 
	[ForeignKey("ProjectBackgroundId")]
	public ProjectBackground ProjectBackground { get; set; } 

	public BackgroundEvent()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/background-event.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectBackground,  vBackgroundEventRecord  } from './'
export class BackgroundEvent {
    constructor(backgroundEvent?: BackgroundEvent  | vBackgroundEventRecord )  {
        let properties = [ "backgroundEventId", "description", "endDate", "isEvent", "startDate", "title", "projectBackgroundId",];
        for (let property of properties)
            if (backgroundEvent && backgroundEvent[property])
                this[property] = backgroundEvent[property];
    }

	backgroundEventId : number =   0 ;

    @maxLength(1000)
	description : string =   undefined;

	endDate : Date =   undefined;

	isEvent : boolean = false ;

    @required()
	startDate : Date =   undefined;

    @required()
    @maxLength(100)
	title : string =   undefined;

    @range(0,2147483647)
	projectBackgroundId : number =   undefined;
	projectBackground : ProjectBackground  ;


}
